pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -351,17 +408,36 @@ func TestProcessResults(t *testing.T) {
 				defaultFS = afero.NewOsFs()
 			}()
 
-			err := processResults(tt.resultMap, option.dir)
-			if tt.expectedErr != "" {
-				require.ErrorContains(t, err, tt.expectedErr)
-			} else {
-				require.NoError(t, err)
-			}
-			// Both test cases above have failed Nodes, hence this file should always be created/
+			antreaInterface := fakeclientset.NewSimpleClientset(&controllerInfo, agentInfo1, agentInfo2)
+			k8sClient := fake.NewSimpleClientset(controllerPod, pod1, pod2)
+			err := processResults(context.TODO(), antreaInterface, k8sClient, tt.resultMap, option.dir)
+			require.NoError(t, err)
 			b, err := afero.ReadFile(defaultFS, filepath.Join(option.dir, "failed_nodes"))
 			require.NoError(t, err)
 			data := string(b)
 			for node, err := range tt.resultMap {
+				tgzFileName := fmt.Sprintf("agent_%s.tar.gz", node)
+				if node == "" {
+					tgzFileName = "controller_node-1.tar.gz"
+				}
+				if err != nil {
+					ok, checkErr := afero.Exists(defaultFS, filepath.Join(option.dir, tgzFileName))
+					require.NoError(t, checkErr)
+					require.True(t, ok, "expected support bundle file %s not found", tgzFileName)
+
+					unpackError := compress.UnpackDir(defaultFS, filepath.Join(option.dir, tgzFileName), option.dir)
+					require.NoError(t, unpackError)
+					expectFileName := "logs/agent/antrea-agent.log"
+					if node == "" {
+						expectFileName = "logs/controller/antrea-controller.log"
+					}
```

Comment:
it would probably be better to have a list of expected files inside the bundle, and then check that it matches the actual list of files by calling `assert.ElementsMatch`. Don't we have more files than that in the fallback bundle?

Reply:
depends on the test pod's containers settings, currently there is only one. 

Well compare all path should be ideal, including all filenames and sub-dirs , i'm not sure this worth the effort. I already added a new function 'UnpackDir', it's only use-case is for this test-case.  More helper functions is needed for this case if we want to have a full compare.

Reply:
> depends on the test pod's containers settings, currently there is only one.

Don't we also have "agentinfo" / "controllerinfo"?

Reply:
updated with more check.